CHANGELOG.md

```diff
@@ -1,9 +1,26 @@
 # Changelog
 
-## [Unreleased](https://github.com/MarquezProject/marquez/compare/0.23.0...HEAD)
+## [Unreleased](https://github.com/MarquezProject/marquez/compare/0.24.0...HEAD)
+
+## [0.24.0](https://github.com/MarquezProject/marquez/compare/0.23.0...0.24.0) - 2022-07-28
```

Comment:
@merobi-hub: We would only update the changelog for `0.24.0` right before the release. The `Unreleased` section can still contain a list of changes (that you've added in this PR) that have yet to be released without an official release.

Reply:
@wslulciuc Thanks. My thinking was that if I opened a PR ahead of time the changelog could be reviewed and revised ahead of the release if needed without losing time

Reply:
Ahh makes sense! I didn't want to merge the PR until the tag / release was out, but looks like we're scheduled for the release today, so let's get this merged!

Reply:
@merobi-hub We'll also want to add a recent fix by @collado-mike (see [`2053`](https://github.com/MarquezProject/marquez/pull/2053))

Reply:
@wslulciuc PR 2053 has been added. Good to go?